src/sections/TransactionInfoSection.tsx

```diff
@@ -115,22 +146,29 @@ const TransactionInfoSection = () => {
                         content={
                           <>
                             <LoadingSpinner style={{ marginRight: 5 }} size={15} />
-                            <span>Unconfirmed</span>
+                            <span>Pending</span>
                           </>
                         }
                       />
                     </td>
                   )}
                 </TableRow>
-                {isTxConfirmed(txInfo) && txInfo.blockHash && (
+                {isTxConfirmed(txInfo) && txInfo.blockHash && txBlock && (
                   <TableRow>
-                    <td>Block Hash</td>
+                    <td>Block</td>
                     <td>
-                      <TightLink
-                        to={`../blocks/${txInfo.blockHash || ''}`}
-                        text={txInfo.blockHash || ''}
-                        maxWidth="550px"
-                      />
+                      <SimpleLink to={`../blocks/${txInfo.blockHash || ''}`}>{txBlock?.height.toString()}</SimpleLink>
+                      <span data-tip="Number of blocks mined since" style={{ marginLeft: 10 }}>
+                        <Badge
+                          type="neutral"
+                          content={
+                            <span>
+                              {confirmations ?? '0'} {confirmations === 1 ? 'Confirmation' : 'Confirmations'}
```

Comment:
With the above change that I propose, confirmations is always a number, so no need to check for undefined.

```suggestion
                              {confirmations} {confirmations === 1 ? 'Confirmation' : 'Confirmations'}
```

Reply:
Done!